frontend/tests/frontend.spec.js

```diff
@@ -0,0 +1,160 @@
+  });
+
+  expect(isPackaged).toBe(false);
+
+  // Evaluation expression in the Electron context.
+  const appPath = await electronApp.evaluate(async ({ app }) => {
+    // This runs in the main Electron process, parameter here is always
+    // the result of the require('electron') in the main app script.
+    return app.getAppPath();
+  });
+
+  console.log(appPath);
+});
+
+test("window title", async () => {
+  // test number of windows
+  const windows = electronApp.windows();
+  expect(windows.length).toBe(1);
+
+  // test window title
+  expect(await window.title()).toBe("DedupliFHIR");
+});
+
+test("UI elements present", async () => {
+  // Check for content
+  const content = await window.locator("div.content");
+  expect(content).not.toBeNull();
+  expect(content.locator("h1")).toHaveText("DedupliFHIR");
+  await window.waitForSelector("h2");
+  expect(content.locator("h2")).toHaveText("Upload Patient Records File");
+
+  // Check for file dropper
+  const fileDropper = await window.locator("input[type='file']");
+  expect(fileDropper).not.toBeNull();
+  expect(fileDropper).toHaveId("file-input-specific");
+  expect(fileDropper).toHaveAttribute("accept", ".json,.csv");
+
+  // Check for dropdown
+  const dropdown = await window.locator("select");
+  expect(dropdown).not.toBeNull();
+  expect(dropdown).toHaveId("options");
+  expect(dropdown).toHaveClass("usa-select");
+
+  // Check for submit button
+  const submitButton = await window.locator("button[type='button']");
+  expect(submitButton).not.toBeNull();
+  expect(submitButton).toHaveId("submit");
+});
+
+test("submit without uploading file", async () => {
+  const submitButton = await window.locator("button[type='button']");
+  await submitButton.click();
+
+  const alert = await window.locator("#file-input-alert");
+  expect(alert).not.toBeNull();
+  expect(alert.locator("p")).toHaveText(
+    "File not found. Please upload a file.",
+  );
+});
+
+test("upload file and submit", async () => {
+  // TODO: remove this when height bug is fixed
+  await window.setViewportSize({ width: 1000, height: 3000 });
+
+  const fileDropper = await window.locator("input[type='file']");
+
+  await fileDropper.setInputFiles("../cli/deduplifhirLib/tests/test_data.csv");
+  // OR, put a custom test file in frontend/tests/
+  // await fileDropper.setInputFiles("tests/test_data.csv");
+
+  const submitButton = await window.locator("button[type='button']");
+  await submitButton.click();
+
+  await window.waitForSelector("div.loader");
+  const loader = await window.locator("div.loader");
+  expect(loader).not.toBeNull();
+});
+
+test("finished results present", async () => {
+  await window.waitForSelector("div.content");
+  const content = await window.locator("div.content");
+  expect(content).not.toBeNull();
+  expect(content.locator("h1")).toHaveText("DedupliFHIR");
+
+  // Wait for results
+  await window.waitForSelector("h2", {
+    timeout: 1000 * 60 * 2, // two minutes
```

Comment:
Hmmmm on my end for this test, it times out at 30 seconds instead of the 2 minutes you put here. Does this happen on your end?

I added the timeout flag `npx playwright test --timeout=120000`, then it works fine and all the tests pass. Maybe we add this flag to the `npm run test` command?

Reply:
The benefit of doing it this way is that we don't have to change the global timeout, but sounds like it's not currently working. I don't feel too strongly about it either way since these run pretty fast regardless